### PR TITLE
proxmox_kvm: recognize force=yes with state=absent

### PR DESCRIPTION
**Pull request by Aversiste**
##### SUMMARY

Recognize `force=yes` in conjunction with `state=absent` to forcibly remove a running VM instead of issuing a warning.

Requires #811.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

proxmox_kvm

##### ADDITIONAL INFORMATION

Using these tasks:

```
- name: "Ensure test-instance is forcibly removed"
  proxmox_kvm:
    api_host: "{{ host }}"
    api_user: "{{ user }}@{{ method }}"
    api_password: "{{ password }}"
    name: test-instance
    state: absent
    force: yes
  register: test

- debug:
    msg: "{{ test }}"
```

Before:

```
TASK [ansible-role-proxmox-instance : Ensure test-instance is removed] ***
ok: [localhost]

TASK [ansible-role-proxmox-instance : debug] ***********************************
ok: [localhost] => 
  msg:
    changed: false
    failed: false
    msg: VM 42 is running. Stop it before deletion.
```

After:

```
TASK [ansible-role-proxmox-instance : Ensure test-instance is removed] ***
changed: [localhost]

TASK [ansible-role-proxmox-instance : debug] ***********************************
ok: [localhost] => 
  msg:
    changed: true
    failed: false
    msg: VM 42 removed
```

### CODE REVIEW

File: changelogs/fragments/811-proxmox-kvm-state-absent.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - proxmox_kvm - defer error-checking for non-existent VMs in order to fix idempotency of tasks using ``state=absent`` and properly recognize a success (https://github.com/ansible-collections/community.general/pull/811).
```
Comment (aminvakil) [inline changelogs/fragments/811-proxmox-kvm-state-absent.yml:3]: I don't think this file is needed as it belongs to another PR.

File: plugins/modules/cloud/misc/proxmox_kvm.py [hunk 2]
```diff
@@ -1083,10 +1095,13 @@ def main():
         try:
             vm = get_vm(proxmox, vmid)
             if not vm:
-                module.exit_json(changed=False, msg="VM %s does not exist" % vmid)
```
Comment (aminvakil) [inline plugins/modules/cloud/misc/proxmox_kvm.py:1086]: Could I ask why this `msg` has been removed?

File: changelogs/fragments/811-proxmox-kvm-state-absent.yml (hunk 1 above)
Comment (Aversiste) [reply changelogs/fragments/811-proxmox-kvm-state-absent.yml:3]: Yes, I just started this branch on top of the one used for PR #811. I will rebase if it gets merged.

File: plugins/modules/cloud/misc/proxmox_kvm.py (hunk 2 above)
Comment (Aversiste) [reply plugins/modules/cloud/misc/proxmox_kvm.py:1086]: Only commit 78d8c03ff871634a194eab1655055ae053cd0be1 is relevant for this particular PR. This particular modification is part of PR #811 will be gone if it gets merged.

File: changelogs/fragments/849-proxmox-kvm-state-absent-force.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+---
+breaking_changes:
+  - proxmox_kvm - recognize force=yes in conjunction with state=absent to forcibly remove a running VM (https://github.com/ansible-collections/community.general/pull/849).
```
Comment (Andersson007) [inline changelogs/fragments/849-proxmox-kvm-state-absent-force.yml:3]:
```suggestion
  - proxmox_kvm - recognize ``force=yes`` in conjunction with ``state=absent`` to forcibly remove a running VM (https://github.com/ansible-collections/community.general/pull/849).
```

**Comment by Andersson007** [comment]
СС @helldorado

File: plugins/modules/cloud/misc/proxmox_kvm.py (hunk 2 above)
Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox_kvm.py:1086]: I've added a`[WIP]` in the PR title to make it clear that it is not done and waits for something (in this case, #811).

Comment (Aversiste) [reply plugins/modules/cloud/misc/proxmox_kvm.py]: Rebased on main branch.

**Review by felixfontein — APPROVED** [review]

**Review by Andersson007 — APPROVED** [review]
LGTM
@felixfontein can we merge this (not backporting to stable-1) so it'll automatically go to the next major release from stable-2?

**Comment by felixfontein** [comment]
@Andersson007 yep. I was planning on adding a message saying that this must not be backported to stable-1 after merging ;)

**Comment by felixfontein** [comment]
@Aversiste thanks for implementing this!
@Andersson007 thanks for reviewing this!

This will **not** get backported to stable-1 since this is a breaking change.
